domain/reports/repository.go

```diff
@@ -26,11 +29,58 @@ func NewFHIRRepository(factory fhir.Factory) *fhirRepository {
 	}
 }
 
+func (repo *fhirRepository) Create(ctx context.Context, customerID int, patientID string, report types.Report) error {
+	if report.Id == "" {
+		report.Id = types.ObjectID(uuid.NewString())
+	}
+	observation, err := convertToFHIR(report)
+	if err != nil {
+		return fmt.Errorf("unable to convert report to FHIR observation: %w", err)
+	}
+	err = repo.factory(fhir.WithTenant(customerID)).CreateOrUpdate(ctx, observation)
+	if err != nil {
+		return fmt.Errorf("unable to write observation to FHIR store: %w", err)
+	}
+	return nil
+}
+
 func renderQuantity(quantity *datatypes.Quantity) string {
 	return fmt.Sprintf("%f %s", *quantity.Value, fhir.FromStringPtr(quantity.Unit))
 }
 
-func convertToDomain(observation *resources.Observation, id string) types.Report {
+func convertToFHIR(report types.Report) (*resources.Observation, error) {
+	if report.Type == "heartRate" {
+		value, err := strconv.ParseFloat(report.Value, 64)
+		if err != nil {
+			return nil, fmt.Errorf("unable to parse report value as float: %w", err)
+		}
+		valueDecimal := datatypes.Decimal(value)
+		observation := &resources.Observation{
+			Domain: resources.Domain{
+				Base: resources.Base{
+					ID:           fhir.ToIDPtr(string(report.Id)),
+					ResourceType: "Observation",
+				},
+			},
+			Code: &datatypes.CodeableConcept{
+				Coding: []datatypes.Coding{{
+					System:  &fhir.LoincCodingSystem,
+					Code:    fhir.ToCodePtr("8893-0"),
+					Display: fhir.ToStringPtr("Heart rate Peripheral artery by Palpation"),
+				}},
+			},
+			Subject: &datatypes.Reference{Reference: fhir.ToStringPtr("Patient/" + string(report.PatientID))},
+			ValueQuantity: &datatypes.Quantity{
+				Value: &valueDecimal,
+			},
+		}
+		return observation, nil
+	}
+
+	return nil, errors.New("unknown report type")
+}
```

Comment:
I'm wondering, should this be in the (to be created) `fhir/zorginzage` package?

Reply:
yes, but lets do that all in one go when this all is finished.